src/metabase/api/pulse.clj

```diff
@@ -117,6 +116,35 @@
   ;; return updated Pulse
   (pulse/retrieve-pulse id))
 
+(api/defendpoint GET "/form_input_refresh"
+  "Provides relevant configuration information and user choices for creating/updating Pulses, can block for quite a long
+  amount of time if the conversations-list or the users-list is huge."
+  []
+  (let [chan-types (-> channel-types
+                       (assoc-in [:slack :configured] (slack/slack-configured?))
+                       (assoc-in [:email :configured] (email/email-configured?)))]
+    {:channels (cond
+                 (when-let [segmented-user? (resolve 'metabase-enterprise.sandbox.api.util/segmented-user?)]
+                   (segmented-user?))
+                 (dissoc chan-types :slack)
+
+                 ;; no Slack integration, so we are g2g
+                 (not (get-in chan-types [:slack :configured]))
+                 chan-types
+
+                 ;; if we have Slack enabled, block until we gather the entire list of conversations + users (can be
+                 ;; quite a while).
+                 :else
+                 (try
+                   (let [[conversations users] (map deref [(future (slack/conversations-list))
+                                                           (future (slack/users-list))])
+                         _                     (slack/slack-cache {:conversations conversations :users users})
```

Comment:
If we only need the `:name` of the channels and users we should only cache those rather than the entire gigantic response that we get back from the Slack API